docs/EN_US/ECLLanguageReference/ECLR_mods/Templ-OPTION.xml

```diff
@@ -530,6 +530,15 @@
               <entry>Upper limit of a CSV line read in MB.</entry>
             </row>
 
+            <row>
+              <entry><emphasis>validateFileType</emphasis></entry>
+
+              <entry>Default: true</entry>
+
+              <entry>If false, the compiler uses the definition in the ECL
+              workunit and ignores the file type in the DFU record.</entry>
```

Comment:
This isn't a compile time option.
It's checked at query run time by the engines.
e.g. when a query is run on hthor and thor and looks up a file, if this option is on (default),
if the published file has a file type, it validates that the ECL used to read it, matches.

NB: the JIRA description should be updated too.

Reply:
Also, there are exceptions to when this check is applied:

1. if the ECL is reading CSV, the check is ignored, since CSV is commonly used to read files of an unknown format.
2. if the ECL is reading a fixed record width flat file, the check is ignored. Similar to (1), there are exceptions when reading a blocks of fixed width records.
3. if the ECL is reading XML or JSON, and there's a mismatch - it will only issue a warning not an error.
I can't remember why, but sometimes folks read XML or JSON back as e.g. CSV.